numpy/_core/src/multiarray/nditer_pywrap.c

```diff
@@ -826,6 +823,11 @@ npyiter_init(NewNpyArrayIterObject *self, PyObject *args, PyObject *kwds)
     npy_free_workspace(op_axes_storage);
     npy_free_workspace(op_axes);
     return res;
+
+  pre_alloc_fail:
```

Comment:
I had meant this to become the fall-through part, i.e., remove the `return res` above as well as the `Py_DECREF(op_in_owned` and `npy_free_cache_dim_obj` (and return `res` below). Am I missing something why this is not possible?

Reply:
no, just didn't think of using a fall-through.